### ISSUE TITLE
WARNING: 'envvar' reference target not found when running 'sphinx-build' to build docs

### ISSUE THREAD

**Issue by hwhsu1231** [fish-shell/fish-shell#10435]
### Problem Description

Recently, I attempted to manually execute the `sphinx-build` command to build the documentation for Fish on the command line. I referred to the following content:

* https://github.com/fish-shell/fish-shell/blob/3.7.0/cmake/Docs.cmake#L15-L28

Based on this, I determined that before running the `sphinx-build` command, I should do two things:

1. Build the `fish_indent` target of CMake to generate `fish_indent` in the binary directory.
2. Prepend the binary directory (output directory of `fish_indent`) to the `PATH` environment variable.

However, when actually executing the `sphinx-build` command, I encountered a bunch of warning messages:

```bash
/home/hwhsu1231/Repo/testing/fish-shell/doc_src/cmds/bind.rst:351: WARNING: 'envvar' reference target not found: VISUAL
/home/hwhsu1231/Repo/testing/fish-shell/doc_src/cmds/bind.rst:351: WARNING: 'envvar' reference target not found: EDITOR
/home/hwhsu1231/Repo/testing/fish-shell/doc_src/cmds/cdh.rst:21: WARNING: 'envvar' reference target not found: dirprev
/home/hwhsu1231/Repo/testing/fish-shell/doc_src/cmds/cdh.rst:21: WARNING: 'envvar' reference target not found: dirnext
/home/hwhsu1231/Repo/testing/fish-shell/doc_src/cmds/fish_svn_prompt.rst:28: WARNING: 'envvar' reference target not found: __fish_svn_prompt_color_revision
/home/hwhsu1231/Repo/testing/fish-shell/doc_src/cmds/fish_svn_prompt.rst:31: WARNING: 'envvar' reference target not found: __fish_svn_prompt_char_separator
```

In contrast, directly using cmake to build the `sphinx-docs` target does not produce these warning messages.

What steps might I be missing before executing the `sphinx-build` command?

### How to Reproduce

1. First, attempt to use cmake to build the sphinx-docs target:

   ```bash
   git clone --depth=1 -b 3.7.0 https://github.com/fish-shell/fish-shell.git
   cd fish-shell
   cmake -S . -B build
   cmake --build build --target sphinx-docs
   ```

2. Attempt to manually execute the sphinx-build command to build the HTML documentation:

   ```bash
   git clean -xfdf
   cmake -S . -B build
   cmake --build build --target fish_indent
   export PATH=$(pwd)/build:$PATH
   LANG=en_US.UTF-8 sphinx-build -j 8 -b html -n doc_src/ build/_build
   ```

### Version and Platform

* OS version: Kubuntu 22.04
* Sphinx version: v7.2.4
* Fish version: 3.7.0

### Logs

[log-sphinx-build-fish-shell-docs-3.7.0-warnings.txt](https://github.com/fish-shell/fish-shell/files/14943967/log-sphinx-build-fish-shell-docs-3.7.0-warnings.txt)

**Comment by faho** [comment]
What this means is that there is no corresponding description for these variables anywhere, so the link doesn't work.

That's fine for these variables.

It won't be shown via cmake because that doesn't turn these warnings on.

You're not doing anything wrong, the docs built fine, just ignore these.

**Comment by hwhsu1231** [comment]
Hello, @faho 

> It won't be shown via cmake because that doesn't turn these warnings on.

But even though I modify the content of `sphinx-docs` target by removing the `-q` argument and then rebuild with cmake, the warning messages won't show up.

I wonder how does fish-shell's cmake configuration suppress those sphinx-build warning messages?

![Screenshot_20240411_175742](https://github.com/fish-shell/fish-shell/assets/71438617/3028e9ac-7247-4d27-9f95-e1954098833b)

**Comment by hwhsu1231** [comment]
Hello, @faho 

I think I found the reason. It's because I added the [`-n`](https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-n) argument in `sphinx-build` command. That's why it would generate  those warning messages.

But the reason why I added this `-n` argument is because I took the following command for reference:

https://github.com/fish-shell/fish-shell/blob/15cd74a3bb8287a0cb9eb0fd1b4aa812dd1210f0/CONTRIBUTING.rst?plain=1#L88

I wonder why fish-shell team added this `-n` argument in the `CONTRIBUTING.rst`?

> That's fine for these variables.

Also, I wonder why it is fine to ignore these variables?
